cmd/rhelemeter-server/main.go

```diff
@@ -0,0 +1,438 @@
+
+	cmd.Flags().BoolVarP(&opt.Verbose, "verbose", "v", opt.Verbose, "Show verbose output.")
+
+	cmd.Flags().StringArrayVar(&opt.Whitelist, "whitelist", opt.Whitelist, "Allowed rules for incoming metrics. If one of these rules is not matched, the metric is dropped.")
+	cmd.Flags().StringVar(&opt.WhitelistFile, "whitelist-file", opt.WhitelistFile, "A file of allowed rules for incoming metrics. If one of these rules is not matched, the metric is dropped; one label key per line.")
+	cmd.Flags().StringArrayVar(&opt.ElideLabels, "elide-label", opt.ElideLabels, "A list of labels to be elided from incoming metrics.")
+	cmd.Flags().Int64Var(&opt.LimitBytes, "limit-bytes", opt.LimitBytes, "The maxiumum acceptable size of a request made to the upload endpoint.")
+	cmd.Flags().Int64Var(&opt.LimitReceiveBytes, "limit-receive-bytes", opt.LimitReceiveBytes, "The maxiumum acceptable size of a request made to the receive endpoint.")
+
+	cmd.Flags().StringVar(&opt.LogLevel, "log-level", opt.LogLevel, "Log filtering level. e.g info, debug, warn, error")
+
+	cmd.Flags().StringVar(&opt.TracingServiceName, "internal.tracing.service-name", "rhelemeter-server",
+		"The service name to report to the tracing backend.")
+	cmd.Flags().StringVar(&opt.TracingEndpoint, "internal.tracing.endpoint", "",
+		"The full URL of the trace collector. If it's not set, tracing will be disabled.")
+	cmd.Flags().Float64Var(&opt.TracingSamplingFraction, "internal.tracing.sampling-fraction", 0.1,
+		"The fraction of traces to sample. Thus, if you set this to .5, half of traces will be sampled.")
+	cmd.Flags().StringVar(&opt.TracingEndpointType, "internal.tracing.endpoint-type", string(tracing.EndpointTypeAgent),
+		fmt.Sprintf("The tracing endpoint type. Options: '%s', '%s'.", tracing.EndpointTypeAgent, tracing.EndpointTypeCollector))
+
+	l := log.NewLogfmtLogger(log.NewSyncWriter(os.Stderr))
+	l = log.WithPrefix(l, "ts", log.DefaultTimestampUTC)
+	l = log.WithPrefix(l, "caller", log.DefaultCaller)
+	stdlog.SetOutput(log.NewStdlibAdapter(l))
+	opt.Logger = l
+
+	level.Info(l).Log("msg", "Rhelemeter server initialized.")
+	if err := cmd.Execute(); err != nil {
+		level.Error(l).Log("err", err)
+		os.Exit(1)
+	}
+}
+
+type Options struct {
+	// External server mTLS configuration
+	TLSKeyPath           string
+	TLSCertificatePath   string
+	TLSCACertificatePath string
+
+	// Internal server TLS configuration
+	InternalTLSKeyPath         string
+	InternalTLSCertificatePath string
+
+	OIDCIssuer           string
+	OIDCClientID         string
+	OIDCClientSecret     string
+	OIDCAudienceEndpoint string
+
+	TenantID string
+
+	LabelFlag         []string
+	Labels            map[string]string
+	LimitBytes        int64
+	LimitReceiveBytes int64
+	RequiredLabelFlag []string
+	RequiredLabels    map[string]string
+	Whitelist         []string
+	ElideLabels       []string
+	WhitelistFile     string
```

Comment:
I don't think we are using any of these at the moment. Maybe we can remove them and the respective flags + processing logic for it in `main()`, just to make things simpler?

Reply:
Yes good point. Left them there as we might need them. But doing it the other way around is probably best

Reply:
id let them be, we know this is going to be part of it

Reply:
ok leaving it as is then